### PR TITLE
Make dask_labextension work with SaturnCluster

### PR DESCRIPTION
**Pull request by jsignell**
I am still working on how to test this locally. But this should allow dask-labextension to work with SaturnCluster. dask-labextension sets all clusters to asynchronous, so we need the `__await__` method. I am less sure about the `start` method.

Related PRs: saturncloud/images#34, https://github.com/saturncloud/saturn/pull/398

### CODE REVIEW

File: dask_saturn/core.py [hunk 1]
```diff
@@ -50,6 +52,16 @@ def scheduler_address(self):
...
 
+    def __await__(self):
+        async def _():
+            async with self.lock:
+                if self.status == "created":
+                    await self.start()
+                    assert self.status == "running"
+            return self
+
+        return _().__await__()
```
Comment (jameslamb) [inline dask_saturn/core.py:63]: This looks identical to the existing `__await__()` on `SpecCluster` (https://github.com/dask/distributed/blob/f748d7d539df0285b46b3c966c587dfd76bf3e48/distributed/deploy/spec.py#L46). Aren't we already getting it for free?

Comment (jsignell) [reply dask_saturn/core.py:63]: The `__await__` that is defined on `SpecCluster` is actually further down (https://github.com/dask/distributed/blob/15591929f3a6b7b390a7a5394e1f53fe6a6c16f4/distributed/deploy/spec.py#L361-L371) and includes calls to `self.scheduler` and `self.workers` both of which we are not implementing

Comment (jsignell) [reply dask_saturn/core.py:63]: No worries, I was like wait is that what I did?!

**Comment by jsignell** [comment]
I am marking this as a draft since there are some ongoing issues with it that probably will need changes upstream. I don't anticipate merging it very soon. 

Some issues:
1. What should happen when you click "Create New Cluster" and you already have one? As of the last commit, you get an error message which uses the WeakrefSet on SpecCluster to fail if more than one instance is created.
2. When should a cluster be closed? If you restart a kernel, should a dask_cluster created in that notebook be closed?
3. If the cluster is started in the side panel (using "NEW") and the code injection button is used to add the cluster to the notebook context, the client is initialized with the scheduler address rather than a cluster object, so the auto-detected dashboard_url is wrong. 

Potential path forward:
1. Make it possible in dask-labextension to disable the cluster creation part of the side-panel.

**Pull request closed without merging**